### DISCUSSION TITLE
ImageField not populate width/height attributes

### DISCUSSION THREAD

**Discussion by roman-rr** [encode/django-rest-framework#8134] [General] [answered]
Create an object with ImageField within REST API are not automatically populated values into `width_field` and `height_field`. 
Create the same object from django admin - width and heights are properly passed. 

I dig deeper issue and found following things. 
Dimensions assigned in [ImageFileDescriptor](https://github.com/django/django/blob/main/django/db/models/fields/files.py#L342) setter. 
Setter `__set__` from **ImageFileDescriptor** function executed only once, right after model execute `.save()` method. 
When create object from django admin, this setter executed few times before `.save()` method. 

Currently i solve this issue with hack: 
```python
class Photo(models.Model):
  photo = models.ImageField(
        height_field='photo_height',
        width_field='photo_width',
  )

  class Meta:
        ordering = ('-created_at',)

  def save(self, *args, **kwargs):
        # add additional setter call before save model to re-assign width/height into our fields
        self.photo = self.photo 
        return super(CommentPhotoRel, self).save(*args, **kwargs)
```

But i believe that there are some reasons, setter called only one time after save, rather than admin model creation, where setter called few times before save.

Any thoughts ?

## Checklist
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.

**Comment by roman-rr** [comment]
Hey @tomchristie 
I'm not a pro with Django and Django Rest. 
So please, confirm that issue reproduced with clear django-restframework project, otherwise i will looking more what's wrong on my side.

**Reply by lovelydinosaur** [reply]
Hi Roman, I'm not always available on a 1:1 basis.
We could do with finding better ways to get the community involved on the discussions group here, but I'm not sure how best to do that.

**Reply by roman-rr** [reply]
@tomchristie Just anyone else besides me, who can create a blank django-rest-framework project with ImageField model and check if width and height is being populated properly to database, on POST request with from-data. 
Would be great!

**Comment by roman-rr** [comment, marked as answer]
@tomchristie I found that issue is reproduced only if the image model inherit parent NOT abstract model. In other cases is all fine. Will try to talk with imagekit team. Can be closed as discussion.
